src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/EndToEndTests/EndToEndTests.cs

```diff
@@ -0,0 +1,129 @@
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+                #region Start Process
+                // Arrange
+                Instance template = new Instance
+                {
+                    InstanceOwner = new InstanceOwner { PartyId = instanceOwnerId.ToString() }
+                };
+                string expectedCurrentTaskName = "Task_1";
+
+                // Act
+                string url = $"/{org}/{app}/instances/";
+
+                HttpResponseMessage response = await client.PostAsync(url, new StringContent(template.ToString(), Encoding.UTF8, "application/json"));
+                response.EnsureSuccessStatusCode();
+
+                Instance createdInstance = JsonConvert.DeserializeObject<Instance>(await response.Content.ReadAsStringAsync());
+                instanceGuid = createdInstance.Id.Split('/')[1];
+                dataGuid = createdInstance.Data.Where(d => d.DataType.Equals("default")).Select(d => d.Id).First();
+
+                // Assert
+                Assert.Equal(expectedCurrentTaskName, createdInstance.Process.CurrentTask.ElementId);
+                #endregion
+
+                #region Upload invalid attachment type  
+                // Act
+                url = $"/{org}/{app}/instances/{instanceOwnerId}/{instanceGuid}/data?dataType=invalidDataType";
+                response = await client.PostAsync(url, null);
+
+                // Assert
+                Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+                #endregion
+
+                #region Move process to next step
+                // Arrange
+                string expectedNextTask = "Task_2";
+                url = $"/{org}/{app}/instances/{instanceOwnerId}/{instanceGuid}/process/next";
+
+                // Act
+                response = await client.GetAsync(url);
+
+                List<string> nextTasks = JsonConvert.DeserializeObject<List<string>>(await response.Content.ReadAsStringAsync());
+                string actualNextTask = nextTasks[0];
+
+                // Assert
+                Assert.Equal(expectedNextTask, actualNextTask);
+
+                // Act
+                response = await client.PutAsync(url, null);
+
+                // Assert
+                response.EnsureSuccessStatusCode();
+                #endregion  
+
+                #region Upload form data during Task_2
+                // Arrange
+                url = $"/{org}/{app}/instances/{instanceOwnerId}/{instanceGuid}/data/{dataGuid}";
+
+                // Act
+                response = await client.PutAsync(url, null);
+
+                // Assert: Upload for data during step 2
+                Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
+                #endregion
+
+                #region Complete process
+                //Arrange
+                url = $"/{org}/{app}/instances/{instanceOwnerId}/{instanceGuid}/process/completeProcess";
```

Comment:
@nkylstad @acn-sbuad  er det completeProcesss vi også bruker der. Trodde kanskje vi brukte next også her. @SandGrainOne  noen tanker?

Reply:
Hva mener du med der? Tenker du på hva app frontend egentlig kaller?

Reply:
Ja, jeg tenkte på det.  Kikket litt på koden og det er vel kanskje slik i praksis at det ikke har noen betydning om det er next eller complete process som kjøres her. Enig? For Task1 så er det viktig at frontend ikke trykker complete process. For det tipper jeg den kanskje gjør nå. For slik koden er nå så vil den vel kjøre ferdig hele prosessen.  Dette kan vi diskutere hvordan vi gjør det med frontend.

Reply:
Mener vi gikk fra å bruke completeProcess til å bruke next før jul i frontend.

Reply:
Man må kanskje kunne velge selv når man setter opp skjemaet hva "Knapp" skal gjøre. Enten neste eller send inn skjema og basert på dette kunne kalle rett endepunkt? 
Det er next som kalles når man trykker på send inn
![image](https://user-images.githubusercontent.com/47737608/76214967-0881a600-620e-11ea-8a4b-7513aa935726.png)

Reply:
Vi har noe teoritiske eksempler i fremtiden med f.eks betinget signering. Da kan det få betydning om man trykker next eller complete.  Men det kan kanskje landes. Senere kan man kanskje lage tester hvor vi bruker next. Ref diskusjon om validering så må vi nok inn med noen endringer her.